File: RepoTransBench/target_projects/Python/Java/tusen-ai_MV2D/src/test/java/com/tusenai/mv2d/original/ConfigsDataFullCoverageTest.java
package com.tusenai.mv2d.original;

import com.tusenai.mv2d.ConfigLoader;
import com.google.gson.*;
import org.junit.jupiter.api.*;

import java.io.IOException;
import java.util.*;

import static org.junit.jupiter.api.Assertions.*;

public class ConfigsDataFullCoverageTest {

    @Test
    public void testDataSampleKeysAndPipelines() throws IOException {
        JsonObject mod = ConfigLoader.loadConfig("configs/mv2d/data/single_frame.json");
        JsonArray classNames = mod.getAsJsonArray("class_names");
        boolean hasCar = false;
        for (JsonElement e : classNames) {
            if (e.getAsString().equalsIgnoreCase("car")) hasCar = true;
        }
        assertTrue(hasCar, "car must be in class_names");

        Set<String> allowedKeys = new HashSet<>(Arrays.asList(
            "type", "to_float32", "with_bbox_3d", "with_label_3d", "with_bbox_2d",
            "with_attr_label", "point_cloud_range", "classes", "data_aug_conf",
            "training", "rot_range", "translation_std", "scale_ratio_range",
            "reverse_angle", "debug", "keys", "class_names", "mean", "std",
            "p", "keep_shape", "imdecode_backend", "size", "keep_ratio", "pad_val",
            "to_rgb", "color_type", "to_onehot", "file_client_args", "backend", "crop_size",
            "size_divisor"
        ));

        JsonArray trainPipeline = mod.getAsJsonArray("train_pipeline");
        JsonArray testPipeline = mod.getAsJsonArray("test_pipeline");

        for (JsonElement el : trainPipeline) {
            assertTrue(el.isJsonObject());
            assertTrue(el.getAsJsonObject().has("type"));
            assertTrue(el.getAsJsonObject().get("type").isJsonPrimitive());
            assertTrue(el.getAsJsonObject().get("type").getAsJsonPrimitive().isString());
            for (Map.Entry<String, JsonElement> e : el.getAsJsonObject().entrySet()) {
                assertTrue(allowedKeys.contains(e.getKey()), "Unexpected key " + e.getKey());
            }
        }
        for (JsonElement el : testPipeline) {
            assertTrue(el.isJsonObject());
            assertTrue(el.getAsJsonObject().has("type"));
            assertTrue(el.getAsJsonObject().get("type").isJsonPrimitive());
            assertTrue(el.getAsJsonObject().get("type").getAsJsonPrimitive().isString());
        }

        JsonObject idaAugConf = mod.getAsJsonObject("ida_aug_conf");
        for (String k : Arrays.asList("resize_lim", "final_dim", "H", "W", "rand_flip")) {
            assertTrue(idaAugConf.has(k));
        }
    }

    @Test
    public void testInputModalityFields() throws IOException {
        JsonObject mod = ConfigLoader.loadConfig("configs/mv2d/data/single_frame.json");
        JsonObject inputModality = mod.getAsJsonObject("input_modality");
        for (String k : inputModality.keySet()) {
            assertTrue(inputModality.get(k).isJsonPrimitive());
            assertTrue(inputModality.get(k).getAsJsonPrimitive().isBoolean());
        }
    }

    @Test
    public void testDataDictContent() throws IOException {
        JsonObject mod = ConfigLoader.loadConfig("configs/mv2d/data/single_frame.json");
        JsonObject data = mod.getAsJsonObject("data");
        for (String subset : Arrays.asList("train", "val")) {
            assertTrue(data.has(subset));
            JsonObject cfg = data.getAsJsonObject(subset);
            assertTrue(cfg.has("type"));
            assertTrue(cfg.has("data_root"));
            assertTrue(cfg.has("pipeline"));
            assertTrue(cfg.get("classes").isJsonArray());
            assertTrue(cfg.has("ann_file"));
            assertTrue(cfg.get("ann_file").isJsonPrimitive());
            if (cfg.has("ann_file_2d")) {
                assertTrue(cfg.get("ann_file_2d").isJsonPrimitive());
                assertTrue(cfg.get("ann_file_2d").getAsString().endsWith(".json"));
            }
            if (cfg.has("test_mode")) {
                boolean t = cfg.get("test_mode").getAsBoolean();
                assertTrue(t || !t);
            }
        }
    }

    @Test
    public void testPostPointCloudRange() throws IOException {
        JsonObject mod = ConfigLoader.loadConfig("configs/mv2d/data/single_frame.json");
        JsonArray pcr = mod.getAsJsonArray("point_cloud_range");
        assertEquals(6, pcr.size());
    }

    @Test
    public void testTwoFramesDataConfig() throws IOException {
        JsonObject mod = ConfigLoader.loadConfig("configs/mv2d/data/two_frames.json");
        assertTrue(mod.has("class_names"));
        assertTrue(mod.get("class_names").isJsonArray());
        assertTrue(mod.has("train_pipeline") && mod.get("train_pipeline").isJsonArray());
        assertTrue(mod.has("test_pipeline") && mod.get("test_pipeline").isJsonArray());
        assertTrue(mod.has("data") && mod.get("data").isJsonObject());
        JsonObject data = mod.getAsJsonObject("data");
        assertTrue(data.has("train") && data.has("val"));
        JsonArray tp = mod.getAsJsonArray("train_pipeline");
        JsonArray ttp = mod.getAsJsonArray("test_pipeline");
        for (JsonElement el : tp) {
            assertTrue(el.isJsonObject());
            assertTrue(el.getAsJsonObject().has("type"));
        }
        for (JsonElement el : ttp) {
            assertTrue(el.isJsonObject());
            assertTrue(el.getAsJsonObject().has("type"));
        }
    }

    @Test
    public void testTwoFramesInputModality() throws IOException {
        JsonObject mod = ConfigLoader.loadConfig("configs/mv2d/data/two_frames.json");
        JsonObject inputModality = mod.getAsJsonObject("input_modality");
        for (String k : inputModality.keySet()) {
            assertTrue(inputModality.get(k).isJsonPrimitive());
            assertTrue(inputModality.get(k).getAsJsonPrimitive().isBoolean());
        }
    }

    @Test
    public void testTwoFramesPointCloudRange() throws IOException {
        JsonObject mod = ConfigLoader.loadConfig("configs/mv2d/data/two_frames.json");
        JsonArray pcr = mod.getAsJsonArray("point_cloud_range");
        assertEquals(6, pcr.size());
    }
}

File: RepoTransBench/target_projects/Python/Go/tusen-ai_MV2D/tests/original/configs_data_full_coverage_test.go
package original

import (
	"strings"
	"testing"

	"github.com/stretchr/testify/assert"
	"github.com/tusanai/mv2d/tests"
)

func TestDataSampleKeysAndPipelines(t *testing.T) {
	mod, err := tests.LoadPythonModule("configs/mv2d/data/single_frame.py")
	assert.NoError(t, err)
	classNames := mod["class_names"].([]interface{})
	foundCar := false
	for _, v := range classNames {
		if v.(string) == "car" {
			foundCar = true
		}
	}
	assert.True(t, foundCar)

	allowedKeys := map[string]bool{
		"type": true, "to_float32": true, "with_bbox_3d": true, "with_label_3d": true, "with_bbox_2d": true,
		"with_attr_label": true, "point_cloud_range": true, "classes": true, "data_aug_conf": true,
		"training": true, "rot_range": true, "translation_std": true, "scale_ratio_range": true,
		"reverse_angle": true, "debug": true, "keys": true, "class_names": true, "mean": true, "std": true,
		"p": true, "keep_shape": true, "imdecode_backend": true, "size": true, "keep_ratio": true,
		"pad_val": true, "to_rgb": true, "color_type": true, "to_onehot": true, "file_client_args": true,
		"backend": true, "crop_size": true, "size_divisor": true,
	}
	fullPipeline := append(mod["train_pipeline"].([]interface{}), mod["test_pipeline"].([]interface{})...)

	for _, pipeRaw := range fullPipeline {
		pipe := pipeRaw.(map[string]interface{})
		_, ok := pipe["type"].(string)
		assert.True(t, ok)
	}
	idaAugConf := mod["ida_aug_conf"].(map[string]interface{})
	for _, k := range []string{"resize_lim", "final_dim", "H", "W", "rand_flip"} {
		assert.Contains(t, idaAugConf, k)
	}
	for _, stageRaw := range mod["train_pipeline"].([]interface{}) {
		stage := stageRaw.(map[string]interface{})
		for key := range stage {
			assert.Truef(t, allowedKeys[key], "Unexpected key '%s' in pipeline stage: %+v", key, stage)
		}
	}
}

func TestInputModalityFields(t *testing.T) {
	mod, err := tests.LoadPythonModule("configs/mv2d/data/single_frame.py")
	assert.NoError(t, err)
	inputMod := mod["input_modality"].(map[string]interface{})
	for _, v := range inputMod {
		_, ok := v.(bool)
		assert.True(t, ok)
	}
}

func TestDataDictContent(t *testing.T) {
	mod, err := tests.LoadPythonModule("configs/mv2d/data/single_frame.py")
	assert.NoError(t, err)
	for _, subset := range []string{"train", "val"} {
		assert.Contains(t, mod["data"], subset)
		cfg := mod["data"].(map[string]interface{})[subset].(map[string]interface{})
		assert.Contains(t, cfg, "type")
		assert.Contains(t, cfg, "data_root")
		assert.Contains(t, cfg, "pipeline")
		assert.IsType(t, []interface{}{}, cfg["classes"])
		assert.Contains(t, cfg, "ann_file")
		assert.IsType(t, "", cfg["ann_file"])
		if af2d, ok := cfg["ann_file_2d"]; ok {
			str := af2d.(string)
			assert.IsType(t, "", str)
			assert.True(t, strings.HasSuffix(str, ".json"))
		}
		if tm, ok := cfg["test_mode"]; ok {
			b, isBool := tm.(bool)
			assert.True(t, isBool)
			assert.True(t, b == true || b == false)
		}
	}
}

func TestPostPointCloudRange(t *testing.T) {
	mod, err := tests.LoadPythonModule("configs/mv2d/data/single_frame.py")
	assert.NoError(t, err)
	pcRange := mod["point_cloud_range"].([]interface{})
	assert.Len(t, pcRange, 6)
}

func TestTwoFramesDataConfig(t *testing.T) {
	mod, err := tests.LoadPythonModule("configs/mv2d/data/two_frames.py")
	assert.NoError(t, err)
	_, ok := mod["class_names"].([]interface{})
	assert.True(t, ok)
	assert.IsType(t, []interface{}{}, mod["train_pipeline"])
	assert.IsType(t, []interface{}{}, mod["test_pipeline"])
	data := mod["data"].(map[string]interface{})
	_, trainOk := data["train"]
	_, valOk := data["val"]
	assert.True(t, trainOk && valOk)
	fullPipeline := append(mod["train_pipeline"].([]interface{}), mod["test_pipeline"].([]interface{})...)
	for _, pipeRaw := range fullPipeline {
		pipe := pipeRaw.(map[string]interface{})
		_, ok := pipe["type"]
		assert.True(t, ok)
	}
}

func TestTwoFramesInputModality(t *testing.T) {
	mod, err := tests.LoadPythonModule("configs/mv2d/data/two_frames.py")
	assert.NoError(t, err)
	inputMod := mod["input_modality"].(map[string]interface{})
	for _, v := range inputMod {
		_, ok := v.(bool)
		assert.True(t, ok)
	}
}

func TestTwoFramesPointCloudRange(t *testing.T) {
	mod, err := tests.LoadPythonModule("configs/mv2d/data/two_frames.py")
	assert.NoError(t, err)
	pcRange := mod["point_cloud_range"].([]interface{})
	assert.Len(t, pcRange, 6)
}